utils/build-script-impl

```diff
@@ -1968,6 +1972,17 @@ for host in "${ALL_HOSTS[@]}"; do
                     -DCMAKE_SHARED_LINKER_FLAGS:STRING="-fuse-ld=gold"
                 )
             fi
+        elif [[ "${USE_LLD_LINKER}" ]]; then
+            echo "${product}: using lld linker"
+            if [[ "${product}" != "swift" ]]; then
+                # All other projects override the linker flags to add in
+                # lld linker support.
```

Comment:
Why doesn't this apply to LLVM and LLDB?

Reply:
Okay, I see your comment there: they *would* be using LLVM_ENABLE_LLD. So what's special about Swift here? It seems to me that Swift should be built the same way as LLVM, and that products like Foundation are more likely to need something like this. But Swift as a lone special case doesn't make sense, right?

Reply:
Ah yah, I'm not sure why it's not. I was cargo-culting the pattern that was used for -fuse-ld=gold above. Spit balling but I had to venture a guess I think it has to something to do with the way the stdlibs choose the linker they want to use?

Reply:
I even copied the comment and just replaced ```gold``` for ```lld``` from a few lines up. I didn't spend any time understanding too deeply since it seemed like someone had already reasoned about all the issues and didn't see any issue since lld is a drop in for gold.

Reply:
Yeah, I looked closer at the logic and it seems that because we allow SDKs in cmake to override and use specific linkers whomever implemented it decided not don't override the default cmake linker universally when building swift. In practice (at least anyone using build-util and not cmake directly) we don't have a way to actually override the per SDK linker used so things end up getting triggered off of SWIFT_ENABLE_GOLD_LINKER or SWIFT_ENABLE_LLD_LINKER. 

From the comment below:
```
# Swift will selectively use the gold or lld linker on all
# parts except building the standard library.  We
# let the Swift cmake setup figure out how to apply
# that.
```

Reply:
I'm actually fixing that second part as part of an upcoming PR to fix cross compiling from a darwin toolchain to linux/bsd/fuchsia where you can set the linker used just for stdlib:

```
--use-lld-for-stdlib-targets=linux-x86_64,linux-aarch64,fuchsia-x86_64,fuchsia-aarch64
--use-gold-for-stdlib-targets=android-armv7
```

This translates into a bunch of specific SDK linker overrides in cmake (may need some help what to actually name those args).

Reply:
So, the reason that swift is special is because there are certain relocations (IIRC, in the protocol conformance table) which binutils would incorrectly create the wrong type of relocation for the standard library.  gold's behavior was more acceptable and that is why we ended up forcing gold (as too many cases eventually broke).  There is no strong reason to force a particular linker, and personally, I find having the option to be the wrong thing.  We should instead inform the user to adjust their linker flags IMO with the same flag.